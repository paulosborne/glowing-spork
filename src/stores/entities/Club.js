import { types } from 'mobx-state-tree';
import { match } from 'postcode';

const Club = types
  .model('Club', {
    id: types.identifierNumber,
    name: types.string,
    shortName: types.string,
    crestUrl: types.string,
    address: types.string,
    founded: types.maybeNull(types.number),
    clubColors: types.string,
    venue: types.string,
  })
  .views(self => ({
    get postcode() {
      return match(self.address)[0];
    },
  }));

export default Club;
