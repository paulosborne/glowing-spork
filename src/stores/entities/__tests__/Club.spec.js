import { isStateTreeNode, getSnapshot } from 'mobx-state-tree';
import Club from '../Club';
import MockClub from 'MockClubResponse';

describe('Club', () => {
  let instance;

  beforeEach(() => {
    instance = Club.create(MockClub);
  });

  it('should create an instance of a team correctly', () => {
    expect(isStateTreeNode(instance)).toEqual(true);
    expect(getSnapshot(instance)).toMatchSnapshot();
  });

  describe('get postcode', () => {
    it('should return the address correctly', () => {
      expect(instance.postcode).toEqual('SE25 6PU');
    });
  });
});
