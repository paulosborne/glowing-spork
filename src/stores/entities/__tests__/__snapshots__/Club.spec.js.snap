// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Club should create an instance of a team correctly 1`] = `
Object {
  "address": "Whitehorse Lane London SE25 6PU",
  "clubColors": "Red / Blue",
  "crestUrl": "https://crests.football-data.org/354.svg",
  "founded": 1905,
  "id": 354,
  "name": "Crystal Palace FC",
  "shortName": "Crystal Palace",
  "venue": "Selhurst Park",
}
`;
